pallets/process-validation/src/lib.rs

```diff
@@ -193,6 +206,34 @@ pub mod pallet {
 
     // helper methods
     impl<T: Config> Pallet<T> {
+        pub fn restriction_over_max_depth(
+            restriction: Restriction<
+                T::RoleKey,
+                T::TokenMetadataKey,
+                T::TokenMetadataValue,
+                T::TokenMetadataValueDiscriminator
+            >,
+            mut count: u8,
+            max_depth: u8
+        ) -> bool {
+            if count > max_depth {
+                return true;
+            }
+
+            match restriction {
+                Restriction::BooleanBinary {
+                    operator: _,
+                    restriction_a,
+                    restriction_b
+                } => {
+                    count += 1;
```

Comment:
instead of having a mutible `count`it might be better to assign a new variable here to `count + 1`.

Reply:
thanks, removed the `clone()`s too